Backend/src/db/data_version.py

```diff
@@ -0,0 +1,194 @@
+from ..types import VersionName, version_name_to_number, highest_version_number, get_version_name_exponent
+from typing import Self
+from .getters import get_latest_version_number
+
+
+class DataVersion:
+    """
+    Class to handle the data version, if only `dataset_name` is given, the latest version will be taken.
+    """
+
+    def __init__(
+        self,
+        dataset_name: str,
+        version_number: int | None = None,
+        version_name: VersionName | None = None,
+        version_name_number: int | None = None,
+        label: bool = False,
+    ) -> None:
+        self.dataset_name = dataset_name
+
+        if version_number is not None:
+            self.current_version = version_number
+        elif version_name is None:
+            # Just taking the latest version, no arguments given
+            self.current_version: int = self.latest_version
+        else:
+            # Another version then the default
+            self.set_version(version_name, version_number, label)
+
+    def set_version(
+        self, version_name: VersionName, version_name_number: int | None = None, label: bool = False
+    ) -> None:
+        """
+        To get the latest version of a collection, it is more efficient to be able to sort by a number.
+        This will set value of `self.set_version`
+
+        0 000000 0 (Format of versions)
+        0 - this is version as in the version_name
+          000000 - this is the version number (The iteration of the model or cluster..., max iterations is 999999)
+                 0 - this is a indicator if it is labeled or not
+
+        Ex)
+        0 - <filename> (datafile)
+        10000000 - <filename>_cluster_0
```

Comment:
Praise: cleaver implementation for the versioning id.

Reply:
Thanks!